webapp/components/DataQuery/Visualizer/Table/Row/Column/CustomAggregateFunctionsEditor/useCustomAggregateFunctionsEditor.js

```diff
@@ -0,0 +1,85 @@
+import { useState } from 'react'
+import { useDispatch } from 'react-redux'
+
+import { AggregateFunction } from '@core/survey/aggregateFunction'
+import * as NodeDef from '@core/survey/nodeDef'
+
+import { DialogConfirmActions } from '@webapp/store/ui'
+import { NodeDefsActions } from '@webapp/store/survey'
+
+export const useCustomAggregateFunctionsEditor = (props) => {
+  const { nodeDef } = props
+
+  const dispatch = useDispatch()
+  const [editedUuid, setEditedUuid] = useState(null)
+
+  const [customAggregateFunctions, setCustomAggregateFunctions] = useState(NodeDef.getAggregateFunctions(nodeDef))
+
+  const onNew = () => {
+    const newFn = AggregateFunction.newAggregateFunction()
+    setCustomAggregateFunctions([...customAggregateFunctions, newFn])
+    setEditedUuid(newFn.uuid)
+  }
+
+  const onEditCancel = (fn) => {
+    const { uuid, placeholder } = fn
+    if (placeholder) {
+      const functionsUpdated = customAggregateFunctions.filter((f) => f.uuid !== uuid)
+      setCustomAggregateFunctions(functionsUpdated)
+    }
+    setEditedUuid(null)
+  }
+
+  const onSave = (fnUpdated) => {
+    const { placeholder, ...fnToSave } = fnUpdated
+    const fnIndex = customAggregateFunctions.findIndex((fn) => fn.uuid === fnToSave.uuid)
+    const functionsUpdated = [...customAggregateFunctions]
+    functionsUpdated.splice(fnIndex, 1, fnToSave)
+    setCustomAggregateFunctions(functionsUpdated)
+    setEditedUuid(null)
```

Comment:
maybe is better manage some dictionary {uuid: Object } and then use the Objects.values and keys

Reply:
ok ok I'll change it!